esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -122,32 +192,49 @@ void WUDetailsVisitor::noteStatistic(StatisticKind kind, unsigned __int64 value,
 
 void WUDetailsVisitor::noteAttribute(WuAttr attr, const char * value)
 {
-    if (extraAttributesRequested)
+    // Notes regarding List type attributes
+    // For list type attributes, noteAttribute is called with the singular equivalent -e.g.IdDependency rather than IdDependencyList
+    //
+    //  - By default only the List type (e.g. IdDependencyList) - that's the case when allAttributes are requested
+    //  - If IdDependency is specifically requested provide that _instead_ of the IdDependencyList
+    WuAttr listAttr = getListAttribute(attr);
+    bool returnList = (listAttr==WaNone)?false:true;
+    bool returnAttr = true;
+
+    if (extraAttributesRequested || returnAttributeSpecified)
```

Comment:
I don't expect it is wrong, but I can't follow the logic of this code.  That suggests it should be split up/ refactored into some functions.  E.g.,

```
if (include(attr))
   addAttrib
else if (listAttr != None && include(listAttr))
   addMultiListItem();

```